streams/src/main/java/org/apache/kafka/streams/processor/internals/DefaultKafkaClientSupplier.java

```diff
@@ -42,4 +43,14 @@
     public Consumer<byte[], byte[]> getRestoreConsumer(Map<String, Object> config) {
         return new KafkaConsumer<>(config, new ByteArrayDeserializer(), new ByteArrayDeserializer());
     }
+
+    /**
+     * Creates a default StreamsKafkaClient. For now there is no need to expose this in the KafkaClientSupplier interface.
+     *
+     * @param config
+     * @return
+     */
+    public StreamsKafkaClient getStreamKafkaClient(StreamsConfig config) {
```

Comment:
I don't think this is necessary to add here. AFAICT `StreamsKafkaClient` is only used in `InternalTopicManager` and it can just be constructed there